services/121-service/src/registration/services/registrations-import.service.ts

```diff
@@ -262,6 +250,24 @@ export class RegistrationsImportService {
       },
     );
 
+    const referenceIds = savedRegistrations.map(
+      (registration) => registration.referenceId,
+    );
+    const messageContentDetails = {
+      messageTemplateKey: RegistrationStatusEnum.new,
+      messageContentType: MessageContentType.new,
+      message: '',
+    };
+    await this.registrationBulkService.applyRegistrationStatusChangeAndSendMessageByReferenceIds(
+      {
+        referenceIds,
+        programId: program.id,
+        registrationStatus: RegistrationStatusEnum.new,
+        userId,
+        messageContentDetails,
+      },
+    );
```

Comment:
It was a bit of hassle, but I do think the implementation now is pretty clean :)